src/modules/SambaAD.pm

```diff
@@ -49,6 +49,9 @@ use constant {
     FALSE => 0,
 };
 
+# create a dummy smb.conf file only for performing the libnet commands
+my $dummy_conf_file = SCR->Read (".target.tmpdir") . "/dummy-smb.conf";
+SCR->Write (".target.string", $dummy_conf_file, "");
```

Comment:
I don't see this being cleaned up.
Also, there's a potential attack vector here. What if an attacker pre-fills /tmp/dummy-smb.conf with malicious settings (is that where ".target.tmpdir" points to?). You should also check the permissions on the file (make sure it's not world writable). Is there an equivalent to tmpfile() in perl?